File: src/gestorAplicacion/Electronico.java
package src.gestorAplicacion;

import java.io.Serializable;
import java.util.ArrayList;

//Definimos la clase abstracta
public abstract class Electronico implements Serializable {
    private Supermercado supermercado;
    private String nombre;
    private int precio;
    private  int cantidad;

    private String marca;

    protected Electronico (String nombre, int precio, String marca, Supermercado supermercado, int cantidad){
        this.nombre = nombre;
        this.precio = precio;
        this.marca = marca;
        this.supermercado = supermercado;
        this.cantidad = cantidad;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getPrecio() {
        return precio;
    }

    public void setPrecio(int precio) {
        this.precio = precio;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }
    public Supermercado getSupermercado(){
        return this.supermercado;
    }
    public void setSupermercado(Supermercado supermercado){
        this.supermercado = supermercado;
    }

	public  int getCantidad() {
		return cantidad;
	}

	public  void setCantidad(int cantidad) {
		this.cantidad = cantidad;
	}


    
    
}

File: Python/Electronico.py
class Electronico():
    def __init__(self,*args):
        if len(args)==2:
            self.nombre=args[0].nombre
            self.marca=args[0].marca
            self.precio=args[0].precio
            self.cantidad=args[1]
            self.supermercado=args[0].supermercado
        else:
            self.nombre=args[0]
            self.marca=args[1]
            self.precio=args[2]
            self.cantidad=args[3]
            self.supermercado=args[4]